src/lib.rs

```diff
@@ -914,23 +922,29 @@ impl Object {
                 ("height", height, |v:String| v.parse().ok()),
                 ("visible", visible, |v:String| v.parse().ok().map(|x:i32| x == 1)),
                 ("rotation", rotation, |v:String| v.parse().ok()),
-            ],
-            required: [
+                ("template", template_path, |v:String| Some(v)),
                 ("x", x, |v:String| v.parse().ok()),
                 ("y", y, |v:String| v.parse().ok()),
             ],
-            TiledError::MalformedAttributes("objects must have an x and a y number".to_string())
+            required: [],
+            TiledError::MalformedAttributes("attributes of object element are malformed".to_string())
         );
         let v = v.unwrap_or(true);
         let w = w.unwrap_or(0f32);
         let h = h.unwrap_or(0f32);
         let r = r.unwrap_or(0f32);
         let id = id.unwrap_or(0u32);
         let gid = gid.unwrap_or(0u32);
+        let x = x.unwrap_or(0f32);
+        let y = y.unwrap_or(0f32);
         let n = n.unwrap_or(String::new());
         let t = t.unwrap_or(String::new());
         let mut shape = None;
         let mut properties = HashMap::new();
+        let template = match tp {
+            Some(path) => Some(Box::new(Template::new(&path, map_path)?)),
```

Comment:
Since templates could be instantiated a lot of times within the same map, it would make sense to avoid loading the template file repeatedly for each instance. This should be doable by caching the loaded templates using their absolute path.